src/EnergyPlus/Autosizing/HeatingCapacitySizing.cc

```diff
@@ -451,7 +451,7 @@ Real64 HeatingCapacitySizer::size(EnergyPlusData &state, Real64 _originalValue,
 
     this->selectSizerOutput(state, errorsFound);
 
-    if (this->isCoilReportObject && this->curSysNum <= state.dataHVACGlobal->NumPrimaryAirSys) {
+    if (this->isCoilReportObject) {
```

Comment:
Many of these issues. The DOAS indexes are currently appended to the number of air loops (i.e., CurSysNum = 6 for a DOAS with 5 AirLoopHVAC objects). When code uses CurSysNum to access air loop arrays there is an access violation. These subtle changes were made to get this defect file to run.

Reply:
Interesting.  Should we actually use a separate counter for DOAS systems rather than keeping them magically in line with the other air loops?  That's something for later, but just curious how you'd like to proceed in the future.

Reply:
We have CurZoneEqNum, CurSysNum, and CurOASysNum. I think we do need a CurDOASNum. While making this change I was also wondering what the long term fix was. Still not sure if `if (this->isCoilReportObject && this->curDOASNum == 0)` is any better than what is here now.